### PR TITLE
OCPQE-9931 Base Profile - Azure worker marketplace images

### PR DESCRIPTION
**Pull request by MayXuQQ**
[OCPQE-9931](https://issues.redhat.com//browse/OCPQE-9931) Base Profile - Azure worker macketplace images

### CODE REVIEW

**Comment by jianlinliu** [comment]
most looks good to me. 

cc @yunjiang29 for review.

File: ci-operator/step-registry/ipi/conf/azure/workers-marketimage/ipi-conf-azure-workers-marketimage-commands.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+images=$(az vm image list --all --offer ocp-worker --publisher redhat -o tsv --query "[?starts_with(urn,'RedHat:ocp-worker:ocp-worker')].urn")
+mapfile -t marketImages <<< "${images}"
+echo "choice market-place image from:" "${marketImages[@]}"
+imageTotal=${#marketImages[@]}
+echo "Image total: ${imageTotal} "
+selected_image_idx=$((RANDOM % ${imageTotal}))
+image=${marketImages[selected_image_idx]}
+echo "$(date -u --rfc-3339=seconds) - Selected image ${image}"
+IFS=':' read -ra imageInfo <<< "${image}"
+echo "total:" ${#imageInfo[@]} "${imageInfo[@]}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/workers-marketimage/ipi-conf-azure-workers-marketimage-commands.sh:34]: I think this is debug, if no need, we can remove it, or only pint out `"${imageInfo[@]}"`

File: ci-operator/step-registry/ipi/conf/azure/workers-marketimage/ipi-conf-azure-workers-marketimage-commands.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+echo "$(date -u --rfc-3339=seconds) - Selected image ${image}"
+IFS=':' read -ra imageInfo <<< "${image}"
+echo "imageInfo: ${imageInfo[@]}"
+# create a patch with existing resource group configuration
+cat > "${PATCH}" << EOF
+compute:
+- platform:
+    azure:
+      osImage:
+        publisher: redhat
```
[the remark this answers is not in the document]
Comment (MayXuQQ) [reply ci-operator/step-registry/ipi/conf/azure/workers-marketimage/ipi-conf-azure-workers-marketimage-commands.sh:45]: the publisher is case sensitive, just accept "redhat", the urn use "RedHat" can not get the image

**Pull request merged**